src/coreclr/src/System.Private.CoreLib/src/System/Threading/ProcessorIdCache.cs

```diff
@@ -97,23 +97,23 @@ internal static bool ProcessorNumberSpeedCheck()
                 // we will measure at least 1 microsecond,
                 // and use at least 1/2 of ProcID iterations
                 // we assume that TLS can't be more than 2x slower than ProcID
-                iters /= 2;
+                iters /= 4;
                 do
                 {
+                    iters *= 2;
                     t1 = Stopwatch.GetTimestamp();
                     for (int j = 0; j < iters; j++)
                     {
                         UninlinedThreadStatic();
                     }
                     t1 = Stopwatch.GetTimestamp() - t1;
-                    iters *= 2;
                 } while (t1 < oneMicrosecond) ;
 
                 minTLS = Math.Min(minTLS, (double)t1 / iters);
             }
 
-            s_processorIdRefreshRate = Math.Min((int)(minID / minTLS), MaxIdRefreshRate);
-            return s_processorIdRefreshRate >= 3;
+            s_processorIdRefreshRate = Math.Min((int)(minID * 5 / minTLS), MaxIdRefreshRate);
+            return s_processorIdRefreshRate <= 5;
```

Comment:
Should this be 3 * 5 ?

Reply:
we do not need to have precise math here. RDPID machines will have the rate at 2-3. The next tier RDTSCP will be ~10.   
5 is just to separate one from another.
Previous '3' was not precise either.

Reply:
ok